v0/destinations/facebook_offline_conversions/utils.js

```diff
@@ -0,0 +1,379 @@
+  }
+
+  if (!standardEvents.length) {
+    standardEvents.push("Other");
+  }
+
+  return standardEvents;
+};
+
+/**
+ * Returns an array of ids which is mapped with standard event from webapp configuration
+ * @param {*} eventsToIds
+ * @param {*} standardEvent
+ * @returns
+ */
+const getEventSetIds = (eventsToIds, standardEvent) => {
+  let eventSetIds = [];
+  const eventsToIdsMapping = getHashFromArrayWithDuplicate(
+    eventsToIds,
+    "from",
+    "to",
+    false
+  );
+
+  const keys = Object.keys(eventsToIdsMapping);
+  keys.forEach(key => {
+    if (key === standardEvent) {
+      eventSetIds = [...eventsToIdsMapping[key]];
+    }
+  });
+
+  return eventSetIds;
+};
+
+/**
+ * Returns eventSetIds and standard event name
+ * @param {*} destination
+ * @param {*} event
+ * @returns
+ */
+const getStandardEventsAndEventSetIds = (destination, event) => {
+  const payload = [];
+  const { Config } = destination;
+  const { eventsToIds, eventsToStandard } = Config;
+  const eventsMapping = getHashFromArrayWithDuplicate(
+    eventsToStandard,
+    "from",
+    "to",
+    false
+  );
+  const standardEvents = getStandardEvents(eventsMapping, event);
+  standardEvents.forEach(standardEvent => {
+    const eventSetIds = getEventSetIds(eventsToIds, standardEvent);
+    if (eventSetIds.length) {
+      payload.push({ standardEvent, eventSetIds });
+    }
+  });
+  if (!payload.length) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[Facebook Offline Conversions] :: Please Map Your Standard Events With Event Set Ids"
+      )
+      .setStatus(400)
+      .setStatTags({
+        destType: DESTINATION,
+        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+        meta:
+          TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.CONFIGURATION
+      })
+      .build();
+  }
+
+  return payload;
+};
+
+/**
+ * Returns content type
+ * @param {*} message
+ * @returns
+ */
+const getContentType = message => {
+  let contentType = "product";
+  const { event } = message;
+  if (event === "Product List Viewed") {
+    contentType = "product_group";
+  }
+  return contentType;
+};
+
+/**
+ * Returns an array of products
+ * @param {*} contents
+ * @returns
+ */
+const getProducts = contents => {
+  const fbContents = [];
+  const products = [];
+  if (Array.isArray(contents)) {
+    fbContents.push(...contents);
+  } else {
+    fbContents.push(contents);
+  }
+
+  fbContents.forEach(content => {
+    const id = content.product_id || content.id || content.sku;
+    const quantity = content.quantity || 1;
+    const { price, brand, category } = content;
+    if (id) {
+      const obj = { id, quantity, price, brand, category };
+      products.push(removeUndefinedAndNullValues(obj));
+    }
+  });
+  return products;
+};
+
+/**
+ * Returns actionSource value
+ * @param {*} payload
+ * @returns
+ */
```

Comment:
Lets add this ref : https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/server-event#action-source here